lib/rubocop/cop/rspec/verified_double_reference.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for consistent verified double reference style.
+      #
+      # @see https://relishapp.com/rspec/rspec-mocks/docs/verifying-doubles
+      #
+      # This cop can be configured in your configuration using the
+      # `EnforcedStyle` option and supports `--auto-gen-config`.
+      #
+      # @example `EnforcedStyle: const`
+      #   # bad
+      #   let(:foo) do
+      #     instance_double('ClassName', method_name: 'returned_value')
+      #   end
+      #
+      #   # good
+      #   let(:foo) do
+      #     instance_double(ClassName, method_name: 'returned_value')
+      #   end
+      #
+      # @example `EnforcedStyle: string`
+      #   # bad
+      #   let(:foo) do
+      #     instance_double(ClassName, method_name: 'returned_value')
+      #   end
+      #
+      #   # good
+      #   let(:foo) do
+      #     instance_double('ClassName', method_name: 'returned_value')
+      #   end
+      class VerifiedDoubleReference < Base
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<style>s` references for verfied doubles.'
+
+        SUPPORTED_STYLES = %w[const string].freeze
+        RESTRICT_ON_SEND = %i[instance_double instance_spy class_double].freeze
+
+        # @!method verified_double(node)
+        def_node_matcher :verified_double, <<-PATTERN
+          (send nil? {:instance_double :instance_spy :class_double} $_ ...)
```

Comment:
I _think_ that when `RESTRICT_ON_SEND` is set, you don’t need to repeat the method names here – so e.g. `(send nil? _ $_ ...)` should ok. @pirj can you confirm or debunk that idea?

Reply:
Fair enough. This is reasonable to avoid the redundant check with a spec covering that unrelated methods are not flagged.

Reply:
Otherwise, interpolating `RESTRICT_ON_SEND.map(&:inspect).join(' ')` is a way to avoid duplication.

Reply:
@pirj I think it's worth questioning how stable the RESTRICT_ON_SEND behavior is in rubocop. If the feature is part of the public api, then is rubocop responsible for the tests? I imagine it is considered public since it is shown in a development doc example on https://docs.rubocop.org/rubocop/development.html

I guess the downside to my thinking is if and when there's a breaking change in rubocop.

Reply:
For the sake of time, feel free to share a testing approach if you strongly disagree.

My initial thought would be a partial spy on `VerifiedDoubleReference#on_send`

Reply:
I can imagine that RuboCop introduces a new `--skip-performance-optimizations` flag that would not respect `RESTRICT_ON_SEND` when used, without us noticing.

So even though it's mentioned in the public docs

Reply:
Edited :point_up: